pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -708,31 +718,78 @@ func (c *controller) reconcileMachineHealth(ctx context.Context, machine *v1alph
 					machine.Name,
 					timeOutDuration,
 				)
+				// Log the error message for machine failure
+				klog.Error(description)
+
+				clone.Status.LastOperation = v1alpha1.LastOperation{
+					Description:    description,
+					State:          v1alpha1.MachineStateFailed,
+					Type:           machine.Status.LastOperation.Type,
+					LastUpdateTime: metav1.Now(),
+				}
+				clone.Status.CurrentStatus = v1alpha1.CurrentStatus{
+					Phase: v1alpha1.MachineFailed,
+					//TimeoutActive:  false,
+					LastUpdateTime: metav1.Now(),
+				}
+				objectRequiresUpdate = true
 			} else {
-				// Timeour occurred due to machine being unhealthy for too long
+
+				// Timeout occurred due to machine being unhealthy for too long
 				description = fmt.Sprintf(
 					"Machine %s is not healthy since %s minutes. Changing status to failed. Node Conditions: %+v",
 					machine.Name,
 					timeOutDuration,
 					machine.Status.Conditions,
 				)
-			}
 
-			// Log the error message for machine failure
-			klog.Error(description)
+				machineSetName, machineDeployName := getMachineOwnerNames(machine)
```

Comment:
The method `getMachineOwnerNames` uses the naming convention to extract the machine deployment name from the name of the first owner reference, which seems to be the name of the machine set. This is a rather indirect way of access.
If I understand correctly, the machine has a label `name` and its value is the machine deployment name. Wouldn't it be better to use that way? Later the label is used anyway to find pending machines for this deployment.

Reply:
Yes you are right, I'll use the label instead then

Reply:
addressed in 2aa4416f08be317fc00c9b12355fe15ea986b5a8